src/Cache_Command.php

```diff
@@ -20,34 +18,130 @@ class Cache_Command extends WP_CLI_Command {
 	 *
 	 * ## OPTIONS
 	 *
-	 * <post_id>
+	 * [<post_id>]
 	 * : ID of the post to clear the cache for.
 	 *
+	 * [--all]
+	 * : Clear all oEmbed caches.
+	 *
 	 * ## EXAMPLES
 	 *
 	 *     # Clear cache for a post
 	 *     $ wp embed cache clear 123
 	 *     Success: Cleared oEmbed cache.
+	 *
+	 *     # Clear all caches
+	 *     $ wp embed cache clear --all
+	 *     Success: Cleared all oEmbed caches.
 	 */
 	public function clear( $args, $assoc_args ) {
 		/** @var \WP_Embed $wp_embed */
-		global $wp_embed;
+		global $wp_embed, $wpdb;
+
+		$all = \WP_CLI\Utils\get_flag_value( $assoc_args, 'all' );
+
+		$post_id = isset( $args[0] ) ? $args[0] : null;
+
+		if ( null === $all && null === $post_id ) {
+			WP_CLI::error( 'You must specify at least one post id or use --all' );
+		}
+
+		if ( $post_id && $all ) {
+			WP_CLI::error( 'You cannot specify both a post id and use --all' );
+		}
+
+		// Delete all oEmbed caches.
+		if ( $all ) {
+			// Get post meta oEmbed caches
+			$oembed_post_meta_post_ids = (array) $wpdb->get_col(
+				"SELECT DISTINCT post_id FROM $wpdb->postmeta
+				WHERE meta_key REGEXP '^_oembed_[0-9a-f]{32}$'
```

Comment:
Could we define these regular expressions as class-level constants, and link to where they're defined in core?

`0-9a-f]{32}` seems like a magical value, so it'd be good to document where it comes from.